crates/shared/src/arguments.rs

```diff
@@ -228,6 +228,10 @@ pub struct Arguments {
     #[clap(long, env)]
     pub settlement_contract_address: Option<H160>,
 
+    /// Override address of the cow amm product factory contract.
+    #[clap(long, env)]
+    pub cow_amm_product_factory_contract_address: Option<H160>,
```

Comment:
Since if seen this a few times. The CoW AMM contracts are named `CowAmmConstantProduct` not `CowAmmProduct`. The reason is that the underlying invariant that needs to be upheld is encoded as a constant product function. IMO this detail is not very relevant for us and we can just call it CoW AMM. Just wanted to point this this naming error indicates a slight misunderstanding to me.

Reply:
Oh, I see, thanks for the explanation! In that case I'll rename it to `CowAmmConstantProduct`. So then I can implement `CowAmm` struct as a listener who keeps track of the addresses and the tokens (upcoming PR).